cocos/2d/CCSpriteFrameCache.cpp

```diff
@@ -298,6 +298,58 @@ void SpriteFrameCache::addSpriteFramesWithDictionary(ValueMap& dictionary, Textu
     CC_SAFE_DELETE(image);
 }
 
+void SpriteFrameCache::addSpriteFramesWithDictionary(ValueMap& dict, const std::string &texturePath)
+{
+    std::string pixelFormatName;
+    if (dict.find("metadata") != dict.end())
+    {
+        ValueMap& metadataDict = dict.at("metadata").asValueMap();
+        if (metadataDict.find("pixelFormat") != metadataDict.end())
+        {
+            pixelFormatName = metadataDict.at("pixelFormat").asString();
+        }
+    }
+    
+    Texture2D *texture = nullptr;
+    static std::unordered_map<std::string, Texture2D::PixelFormat> pixelFormats = {
+        {"RGBA8888", Texture2D::PixelFormat::RGBA8888},
+        {"RGBA4444", Texture2D::PixelFormat::RGBA4444},
+        {"RGB5A1", Texture2D::PixelFormat::RGB5A1},
+        {"RGBA5551", Texture2D::PixelFormat::RGB5A1},
+        {"RGB565", Texture2D::PixelFormat::RGB565},
+        {"A8", Texture2D::PixelFormat::A8},
+        {"ALPHA", Texture2D::PixelFormat::A8},
+        {"I8", Texture2D::PixelFormat::I8},
+        {"AI88", Texture2D::PixelFormat::AI88},
+        {"ALPHA_INTENSITY", Texture2D::PixelFormat::AI88},
+        //{"BGRA8888", Texture2D::PixelFormat::BGRA8888}, no Image convertion RGBA -> BGRA
+        {"RGB888", Texture2D::PixelFormat::RGB888}
```

Comment:
Why only these pixel format should change Texture2D's default pixel format before creating textures?